cmd/lakectl/cmd/sst.go

```diff
@@ -0,0 +1,180 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	sst "github.com/cockroachdb/pebble/sstable"
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/catalog/rocks"
+	"github.com/treeverse/lakefs/graveler/committed"
+	"github.com/treeverse/lakefs/graveler/sstable"
+)
+
+type InMemFileInfo struct {
+	name string
+	size int64
+}
+
+func (r *InMemFileInfo) Name() string       { return r.name }
+func (r *InMemFileInfo) Size() int64        { return r.size }
+func (r *InMemFileInfo) Mode() os.FileMode  { return os.ModePerm }
+func (r *InMemFileInfo) ModTime() time.Time { return time.Now() }
+func (r *InMemFileInfo) IsDir() bool        { return false }
+func (r *InMemFileInfo) Sys() interface{}   { return nil }
+
+type InMemReadableFile struct {
+	data *bytes.Reader
+}
+
+func (r *InMemReadableFile) ReadAt(p []byte, off int64) (n int, err error) {
+	return r.data.ReadAt(p, off)
+}
+
+func (r *InMemReadableFile) Close() error {
+	return nil
+}
+
+func (r *InMemReadableFile) Stat() (os.FileInfo, error) {
+	return &InMemFileInfo{
+		name: "stdin",
+		size: r.data.Size(),
+	}, nil
+}
+
+func getIterFromFile(filePath string) (committed.ValueIterator, map[string]string, error) {
+	var file sst.ReadableFile
+	var err error
+	// read from stdin (file has to be seekable/stat-able so we have this weird wrapper
+	if filePath == "-" {
+		data, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			return nil, nil, err
+		}
+		file = &InMemReadableFile{
+			bytes.NewReader(data),
+		}
+	} else {
+		file, err = os.Open(filePath)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	// read file descriptor
+	reader, err := sst.NewReader(file, sst.ReaderOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	// create an iterator over the whole thing
+	iter, err := reader.NewIter(nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	// wrap it in a Graveler iterator
+	dummyDeref := func() error { return nil }
+	return sstable.NewIterator(iter, dummyDeref), reader.Properties.UserProperties, nil
+}
+
+func writeRangeSSTable(iter committed.ValueIterator, amount int) *Table {
+	rows := make([][]interface{}, 0)
+	for iter.Next() {
+		if amount != -1 && len(rows)+1 > amount {
+			break
+		}
+		v := iter.Value()
+		gv, err := committed.UnmarshalValue(v.Value)
+		if err != nil {
+			panic(err)
+		}
+		ent, err := rocks.ValueToEntry(gv)
+		if err != nil {
+			panic(err)
+		}
+		rows = append(rows, []interface{}{
+			string(v.Key),
+			fmt.Sprintf("%x", gv.Identity),
+			fmt.Sprint(ent.GetSize()),
```

Comment:
Why convert a number to a string, if it's going to be an `interface{}` anyway?

Reply:
because `go-pretty/table` formats something very weird for numbers if I leave them as is.